src/agent_writer.cpp

```diff
@@ -156,12 +156,35 @@ void AgentWriter::startWriting(std::unique_ptr<Handle> handle) {
             payload = trace_encoder_->payload();
             trace_encoder_->clearTraces();
           }  // lock on mutex_ ends.
-          // Send spans, not in critical period.
+          // Send the spans. Note that this is not in the critical section above.
           bool success = retryFiniteOnFail(
               [&]() { return AgentWriter::postTraces(handle, headers, payload, logger_); });
+          // Sending could fail. If it succeeds, then the HTTP response status
+          // could indicate an error or success.
           if (success) {
-            trace_encoder_->handleResponse(handle->getResponse());
+            const int response_status = handle->getResponseStatus();
+            if (response_status == 0) {
+              const std::string body = handle->getResponse();
+              std::ostringstream diagnostic;
+              diagnostic << "Datadog Agent returned response without an HTTP status and with the "
+                            "following body of length "
+                         << body.size() << ": " << body;
```

Comment:
I suspect the body would be empty, but if it weren't, it could be pretty long.
So maybe just the first 100 bytes or so instead of the entire body?

Reply:
I considered that. I agree that the response will most likely be empty; but if it's not empty, I don't think it will be long. What might it contain? In the 200 case, it's a JSON object containing one entry per root service seen recently by the Agent, but that's not relevant here. In the error case, it might be... a stack trace? This is Go we're talking about. Logging the body is kind of silly, but I don't think that excessive length will be a problem. Let's leave it.